x-pack/plugins/security_solution/public/users/components/all_users/index.tsx

```diff
@@ -0,0 +1,186 @@
+
+const tableType = usersModel.UsersTableType.allUsers;
+
+interface UsersTableProps {
+  users: User[];
+  fakeTotalCount: number;
+  loading: boolean;
+  loadPage: (newActivePage: number) => void;
+  id: string;
+  showMorePagesIndicator: boolean;
+  totalCount: number;
+  type: usersModel.UsersType;
+  sort: SortUsersField;
+  setQuerySkip: (skip: boolean) => void;
+}
+
+export type UsersTableColumns = [
+  Columns<User['name']>,
+  Columns<User['lastSeen']>,
+  Columns<User['domain']>
+];
+
+const rowItems: ItemsPerRow[] = [
+  {
+    text: i18n.ROWS_5,
+    numberOfRow: 5,
+  },
+  {
+    text: i18n.ROWS_10,
+    numberOfRow: 10,
+  },
+];
+
+const UsersTableComponent: React.FC<UsersTableProps> = ({
+  users,
+  totalCount,
+  type,
+  id,
+  fakeTotalCount,
+  loading,
+  loadPage,
+  showMorePagesIndicator,
+  sort,
+  setQuerySkip,
+}) => {
+  const dispatch = useDispatch();
+  const getUsersSelector = useMemo(() => usersSelectors.allUsersSelector(), []);
+  const { activePage, limit } = useDeepEqualSelector((state) => getUsersSelector(state));
+
+  const updateLimitPagination = useCallback(
+    (newLimit) => {
+      dispatch(
+        usersActions.updateTableLimit({
+          usersType: type,
+          limit: newLimit,
+          tableType,
+        })
+      );
+    },
+    [type, dispatch]
+  );
+
+  const updateActivePage = useCallback(
+    (newPage) => {
+      dispatch(
+        usersActions.updateTableActivePage({
+          activePage: newPage,
+          usersType: type,
+          tableType,
+        })
+      );
+    },
+    [type, dispatch]
+  );
+
+  const onSort = useCallback(
+    (criteria: Criteria) => {
+      if (criteria.sort != null) {
+        const newSort = criteria.sort;
+        if (newSort.direction !== sort.direction || newSort.field !== sort.field) {
+          dispatch(
+            usersActions.updateTableSorting({
+              sort: newSort as SortUsersField,
+              tableType,
+            })
+          );
+        }
+      }
+    },
+    [dispatch, sort]
+  );
+  const columns = useMemo(() => getUsersColumns(), []);
+
+  return (
+    <PaginatedTable
+      activePage={activePage}
+      columns={columns}
+      dataTestSubj={`table-${tableType}`}
+      headerCount={totalCount}
+      headerTitle={i18n.USERS}
+      headerUnit={i18n.UNIT(totalCount)}
+      id={id}
+      itemsPerRow={rowItems}
+      limit={limit}
+      loading={loading}
+      loadPage={loadPage}
+      pageOfItems={users}
+      showMorePagesIndicator={showMorePagesIndicator}
+      totalCount={fakeTotalCount}
+      updateLimitPagination={updateLimitPagination}
+      updateActivePage={updateActivePage}
+      sorting={sort}
+      onChange={onSort}
+      setQuerySkip={setQuerySkip}
+    />
+  );
+};
+
+UsersTableComponent.displayName = 'UsersTableComponent';
+
+export const UsersTable = React.memo(UsersTableComponent);
+
+const getUsersColumns = (): UsersTableColumns => [
```

Comment:
shouldn't we place this const before the component definition, with the `rowItems`?